code/modules/detectivework/scanner.dm

```diff
@@ -234,7 +235,16 @@
 	switch(action)
 		if("clear")
 			clear_logs(ui.user)
-			return
+		if("delete")
+			var/reversed_index = params["index"] // Reversing because UI displays new logs at the top
+			var/index = LAZYLEN(log_data) - reversed_index
+			if(!log_data[index])
+				return
+			if(scanner_busy)
+				balloon_alert(ui.user, "scanner busy!")
+				return
+			log_data.Cut(index, index + 1)
+			balloon_alert(ui.user, "log deleted")
```

Comment:
the ability to delete logs throws my original assumption of "you can just use the index because elements in the array are never modified" out of the window, you cannot use the array index as a key anymore :p

this is fine though, i looked it up and keys only log a warning, they dont cause an exception, that means duplicate keys dont crash the UI. from what i can tell it just prevents one of the children from appearing